src/main/java/com/blackwaterpragmatic/webservices/resource/UserResourceResponseBuilder.java

```diff
@@ -0,0 +1,33 @@
+package com.blackwaterpragmatic.webservices.resource;
+
+import com.blackwaterpragmatic.webservices.bean.User;
+import com.google.common.collect.ImmutableMap;
+
+import javax.ws.rs.core.Response;
+
+/**
+ * It's not perfect, but it encapsulates the boiler plate that just makes for a hard read.
+ * There could be a lot more refactoring here obviously, but the gist is we're a bit dry-er now
+ * and the logic for creating generic entity responses is pulled out of the resources.
+ */
```

Comment:
A concrete implementation for the User object, would have error messages, etc, tailored for the UserResource.  The map is weak, but it was just to demonstrate...